### DISCUSSION TITLE
Private Bot

### DISCUSSION THREAD

**Discussion by fraffo386** [innightwolfsleep/llm_telegram_bot#77] [Q&A]
Hi guys, i was thinking that i'd like the bot to be private, i mean that it only answer to me or someone i give the access, i think it can be done by adding a feature in the code that the bot recognize the User ID of the person who's writing, but i really don't know where to do it, in what file should i do it i mean. I'd really appreciate if you could help me, thanks :))

**Comment by innightwolfsleep** [comment]
Added file "telegram_users.txt" for permitted users id's
If empty - permit for all.

**Reply by fraffo386** [reply]
oh i see, thanks, but what do i need to add in the txt?
my user ID or my username? and if i want to add more users i use a comma, a space or i go on a new line?

**Reply by innightwolfsleep** [reply]
Add users id's, one per line. 

If file is empty - permit for everyone. Otherwithe - only for id's in file.

You dont need bot restart after adding.

**Reply by fraffo386** [reply]
oh ok, thank you so much, i appreciate it a lot <3
